README.md

```diff
@@ -30,36 +31,129 @@ Use this package to parse and validate AsyncAPI documents —either YAML or JSON
 
-### Example passing inline AsyncAPI
+## Examples 
 
-```js
-// TBD
+### Example with parsing
+
+```ts
+import { Parser } from '@asyncapi/parser';
+
+const parser = new Parser();
+
+const { document } = await parser.parse(`
+  asyncapi: '2.4.0'
+  info:
+    title: Example AsyncAPI specification
+    version: '0.1.0'
+  channels:
+    example-channel:
+      subscribe:
+        message:
+          payload:
+            type: object
+            properties:
+              exampleField:
+                type: string
+              exampleNumber:
+                type: number
+              exampleDate:
+                type: string
+                format: date-time
+`);
+
+if (document) {
+  // => Example AsyncAPI specification
+  console.log(document.info().title());
+}
 ```
 
-### Example passing a URL
-
-```js
-// TBD
+### Example with validation
+
+```ts
+import { Parser } from '@asyncapi/parser';
+
+const parser = new Parser();
+
+// One of the diagnostics will contain an error regarding an unsupported version of AsyncAPI (2.1.37)
+const diagnostics = await parser.validate(`
+  asyncapi: '2.1.37'
+  info:
+    title: Example AsyncAPI specification
+    version: '0.1.0'
+  channels:
+    example-channel:
+      subscribe:
+        message:
+          payload:
+            type: object
+            properties:
+              exampleField:
+                type: string
+              exampleNumber:
+                type: number
+              exampleDate:
+                type: string
+                format: date-time
+`);
 ```
 
-### Example using Avro schemas
+### [Example using Avro schemas](#custom-schema-parsers)
 
 Head over to [asyncapi/avro-schema-parser](https://www.github.com/asyncapi/avro-schema-parser) for more information.
 
-### Example using OpenAPI schemas
+### [Example using OpenAPI schemas](#custom-schema-parsers)
 
 Head over to [asyncapi/openapi-schema-parser](https://www.github.com/asyncapi/openapi-schema-parser) for more information.
 
-### Example using RAML data types
+### [Example using RAML data types](#custom-schema-parsers)
 
 Head over to [asyncapi/raml-dt-schema-parser](https://www.github.com/asyncapi/raml-dt-schema-parser) for more information.
 
+### [Example with stringify and unstringify parsed document](#stringify)
+
+```ts
+import { Parser, stringify, unstringify } from '@asyncapi/parser';
+
+const parser = new Parser();
+
+const { document } = await parser.parse(`
+  asyncapi: '2.4.0'
+  info:
+    title: Example AsyncAPI specification
+    version: '0.1.0'
+  channels:
+    example-channel:
+      subscribe:
+        message:
+          payload:
+            type: object
+            properties:
+              exampleField:
+                type: string
+              exampleNumber:
+                type: number
+              exampleDate:
+                type: string
+                format: date-time
+`);
```

Comment:
This example is not valid to Spectral because of this rule `asyncapi-operation-operationId`: https://github.com/stoplightio/spectral/blob/297531bbc6f26393d325a101b59f9e8b1f57e21e/packages/rulesets/src/asyncapi/index.ts#L230-L240

Not sure if we really want to keep that rule there. The `operationId` is not mandatory by the spec.

Reply:
I know about some "problems" with Spectral. Now I would leave that spec, because at the end `operationId` won't be mandatory on Spectral side, don't worry.

Reply:
Do you mean that particular spectral rule will be gone? Could you bring more details, please?

Reply:
Changed will be severity from error to warning - then there will be information that it is a good pattern design. Most of the rules from Spectral were made 2 years ago by the Stoplight team itself and a lot of things are not up to date. This week I have a plan to go through all the rules and improve them.

Reply:
> Changed will be severity from error to warning

That makes totally sense to me. I have few concerns regarding this but I'm not gonna add anything extra into this issue, but rather please ping me on the work you plan to do 🙏

Reply:
I will create PR with that in Spectral side, so maybe better will be if you will list everything bugs that you see, hmm?